README.rst

```diff
@@ -75,6 +75,20 @@ and is removed in Python 3. Use ``str(e)`` to access the user-readable
 message. Use ``e.args`` to access arguments passed to the exception.
 
 
+Opinionated warnings
+~~~~~~~~~~~~~~~~~~~~
+
+Those warnings are disabled by default because the opinions they come from are
+controversial.
+
+**B901**: Using ``return x`` in a generator function is syntactically invalid
+in Python 2, but in Python 3 ``return x`` in a generator function means ``raise
+StopIteration(x)``. Users that come from Python 2 may be used to Python 2
+rejecting code that mixes both, so this can lead to situations where it looks
+like ``return x`` makes the function generate an empty sequence. Use the more
```

Comment:
The suggestion to use `raise StopIteration(x)` as an equivalent is invalid. See PEP 479. StopIterations that come out of generators are not transformed into RuntimeErrors, see specifically:
https://www.python.org/dev/peps/pep-0479/#writing-backwards-and-forwards-compatible-code

Since valid users of this functionality are most likely users of coroutines, a suggestion to use native coroutines here would be enough.

Add a check in your unit test that confirms that native coroutines (`async def` with `await` and `return`) don't raise B901 and we're good.